src/Survey/ItechArt.Survey.Foundation/Authentication/Validation/UserValidator.cs

```diff
@@ -0,0 +1,79 @@
+﻿using ItechArt.Common;
+using ItechArt.Survey.DomainModel;
+using ItechArt.Survey.Foundation.Authentication.Abstractions;
+using System.Text.RegularExpressions;
+
+namespace ItechArt.Survey.Foundation.Authentication.Validation;
+
+public class UserValidator
+{
+    public static OperationResult<UserRegistrationErrors> Validate(User user, string password)
```

Comment:
1. Move all contratins (user/pass min/max length) to options - and use Options pattern
2. Move validation to AuthenticationService class